prow/cmd/deck/main.go

```diff
@@ -408,6 +421,144 @@ func handleBadge(ja *jobs.JobAgent) http.HandlerFunc {
 	}
 }
 
+// handleRequestJobViews handles requests to get all available artifact views for a given job
+// it responds with html containing all available views
+// A valid job view url takes the form:
+//
+// /view/{src=<URI-encoded-source-string>, gcs/<bucketname>/<jobprefix>, prowjob/<jobname>/<buildID>}
+//
+// Examples:
+// - /view/prowjob/echo-test/1021530234601607168
+// - /view/gcs/kubernetes-jenkins/logs/ci-kubernetes-e2e-gce-large-performance/121
+// - /view/?src=gs:%2F%2Fkubernetes-jenkins%2Fpr-logs%2Fpull%2Fkubeflow_kubeflow%2F1195%2Fkubeflow-presubmit%2F2558
+func handleRequestJobViews(sg *spyglass.Spyglass, ca *config.Agent) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		setHeadersNoCaching(w)
+		urlPath := r.URL.Path
+		start := time.Now()
+		var src string
+		if strings.Trim(urlPath, "/") != "view" { //TODO (paulangton): config this prefix later
+			urlPath = strings.TrimPrefix(urlPath, "/view/")
+			splitSource := strings.SplitAfterN(urlPath, "/", 2)
+			if len(splitSource) < 2 {
+				http.Error(w, "invalid view url", http.StatusBadRequest)
+				return
+			}
+			srcExt := splitSource[0]
+			srcData := splitSource[1]
+			switch srcExt {
```

Comment:
Seems like you should have separate handlers for these instead of a larger one that demuxes

Reply:
fixed in latest